### DISCUSSION TITLE
Static wrapping to smaller width in Container width: auto

### DISCUSSION THREAD

**Discussion by aaronst** [Textualize/textual#1450] [Q&A] [answered]
Something changed in version `0.6.0` that affects how `Static` (or maybe `Container`) widgets set their width. Here's an example:

```python
from textual.app import App, ComposeResult
from textual.containers import Container
from textual.widgets import Button, Static

class Test(App):
    CSS = """
    Container {
        width: auto;
    }
    """

    def compose(self) -> ComposeResult:
        yield Container(Static("A wider static widget."), Button("button"))

Test().run()
```

Up to and including version `0.5.0`, the `Container` expands so that the `Static` text displays all on one line. However, since version `0.6.0`, the `Static` text now wraps to the width of the `Button`. I think this is a bug, but I wanted to start a discussion first in case this was an intentional change in functionality.

**Comment by willmcgugan** [comment, marked as answer]
The was a change to the auto width calculations, but this is the expected behaviour. If you want the static to not wrap, you can set its width to fit (i.e. `width: auto`).

**Reply by aaronst** [reply]
Thanks Will.
